components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java

```diff
@@ -95,7 +97,13 @@ public int getListenerOrder() {
     }
 
     @Override
-    public void onPreDelete(int i) throws StratosException {
+    public void onPreDelete(int tenantId) throws StratosException {
 
+        try {
+            UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
```

Comment:
Can there be some other UserRecoveryDataStore which is not type JDBCRecoveryDataStore?

Reply:
JDBCRecoveryDataStore is the only UserRecoveryDataStore implementation used in this component. I think it's fine to use JDBCRecoveryDataStore here. WDYT?

Reply:
Thats is the one we supprt in the latest product by default